sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -223,10 +255,24 @@ public void testReadDisplayData() {
   }
 
   @Test
+  public void testReadDisplayDataWithGqlQuery() {
+    DatastoreV1.Read read =  DatastoreIO.v1().read()
+        .withProjectId(PROJECT_ID)
+        .withGqlQuery(GQL_QUERY)
+        .withNamespace(NAMESPACE);
+
+    DisplayData displayData = DisplayData.from(read);
+
+    assertThat(displayData, hasDisplayItem("projectId", PROJECT_ID));
+    assertThat(displayData, hasDisplayItem("gqlQuery", GQL_QUERY));
+    assertThat(displayData, hasDisplayItem("namespace", NAMESPACE));
+  }
+
+  @Test
   @Category(RunnableOnService.class)
```

Comment:
@tgroh @kennknowles I am assuming at some point we will revisit `NeedsRunner` and `RunnableOnService` annotations across the SDK and clean them up wherever necessary?